ScoutSuite/providers/azure/provider.py

```diff
@@ -35,39 +39,55 @@ def __init__(self, project_id=None, organization_id=None,
 
         super(AzureProvider, self).__init__(report_dir, timestamp, services, skipped_services, thread_config)
 
-    def authenticate(self, key_file=None, user_account=None, service_account=None, **kargs):
+    def authenticate(self, key_file=None, user_account=None, service_account=None, azure_cli=None, azure_msi=None,
+                     azure_service_principal=None, azure_file_auth=None, azure_user_credentials=None, **kargs):
         """
-        Implement authentication for the GCP provider
-        Refer to https://google-auth.readthedocs.io/en/stable/reference/google.auth.html.
+        Implements authentication for the Azure provider using azure-cli.
+        Refer to https://docs.microsoft.com/en-us/python/azure/python-sdk-azure-authenticate?view=azure-python.
 
         :return:
         """
 
         try:
-
-            # TODO this is temporary
-            import os
-            # Azure subscription
-            SUBSCRIPTION_ID = os.environ['SUBSCRIPTION_ID']
-            # Tenant ID for your Azure Subscription
-            TENANT_ID = os.environ['TENANT_ID']
-            # Your Service Principal App ID
-            CLIENT = os.environ['CLIENT']
-            # Your Service Principal Password
-            KEY = os.environ['KEY']
-
-            self.aws_account_id = TENANT_ID # TODO this is for AWS
-
-            credentials = ServicePrincipalCredentials(
-                client_id=CLIENT,
-                secret=KEY,
-                tenant=TENANT_ID
-            )
-
-            self.credentials = AzureCredentials(credentials, SUBSCRIPTION_ID)
-
-            return True
-
+            if azure_cli:
+                cli_credentials, self.aws_account_id = get_azure_cli_credentials()  # TODO: Remove aws_account_id
+                self.credentials = AzureCredentials(cli_credentials, self.aws_account_id)
+                print(cli_credentials)
+                return True
+            elif azure_msi:
+                # TODO test
+                credentials = MSIAuthentication()
+                self.aws_account_id = "aaa"
+                self.credentials = AzureCredentials(credentials, self.aws_account_id)
+                return True
```

Comment:
If this is untested, I should we at least inform the user this functionality is experimental.

Reply:
Yep! I still haven't updated the docs! I just wanted to push what I did, I forgot to write that it wasn't ready to pull!

Reply:
Done!